lms/djangoapps/certificates/signals.py

```diff
@@ -79,7 +83,10 @@ def listen_for_passing_grade(sender, user, course_id, **kwargs):  # pylint: disa
         return
     log.info(f'Attempt will be made to generate a course certificate for {user.id} : {course_id} as a passing grade '
              f'was received.')
-    return generate_certificate_task(user, course_id)
+    try:
+        return generate_certificate_task(user, course_id)
+    except CertificateGenerationNotAllowed as e:
+        return
```

Comment:
this structure is strange

generate_certificate_task returns True or throws an exception
this transforms that into returning True or returning None

I would be surprised if callers were depending on the True from the return, so the most sensible thing might be to have generate_certificate_task have no return normally instead of returning true

also applies to the other call points

Reply:
You're right. I'll be fixing this, thanks!